asr1k_neutron_l3/plugins/l3/service_plugins/l3_extension_adapter.py

```diff
@@ -316,10 +321,29 @@ def create_router(self, context, router):
         asr1k_db.RouterAttsDb.ensure(context, result.get('id'))
         return result
 
+    def ensure_default_route_skip_monitoring(self, context, router_id, router):
+        tag_plugin = directory.get_plugin(TAG_PLUGIN_TYPE)
+        custom_default_route_tags = {TAG_SKIP_MONITORING, TAG_DEFAULT_ROUTE_OVERWRITE}
+
+        has_custom_default_route = any(r['destination'] == '0.0.0.0/0' for r in router['routes'])
+        with context.session.begin(subtransactions=True):
+            router_tags = set(tag_plugin.get_tags(context, 'routers', router_id)['tags'])
+
+            has_default_route_tags = custom_default_route_tags.issubset(router_tags)
+
+            if has_default_route_tags != has_custom_default_route:
+                if has_default_route_tags:
+                    router_tags = router_tags - custom_default_route_tags
+                else:
+                    router_tags = router_tags.union(custom_default_route_tags)
```

Comment:
Okay, this took me a bit to understand. Essentialls this is:
```python
if has_default_route_tags and not has_custom_default_route:
    router_tags = router_tags - custom_default_route_tags
elif not has_default_route_tags and has_custom_default_route:
    router_tags = router_tags.union(custom_default_route_tags)
```
so with the tags and no default route, we remove both tags and with default route but no tags we add the tags, right?

Also, shorthand if you like
```python
router_tags |= custom_default_route_tags
```

Reply:
Not sure what you mean here. But `router_tags  |= custom_default_route_tags` is equal to `router_tags .update(custom_default_route_tags)` [(docs)](https://docs.python.org/3/library/stdtypes.html#set`). So that would only work for the second case. Was your intention to replace the whole block with one set operation?

I chose this way 2 level branch to disinguish clearly between _is there something to change?_ and then _what will I change?_

Reply:
Yes, `router_tags |= custom_default_route_tags` is equal to the statement in your code. I wanted to offer this as a shorthand, as I find it easier to parse in my head. `x |= y` tells me "add y to x" and the same would go for `router_tags -= custom_default_route_tags`.

Okay, I get why you chose to use the multilevel™ if-clause. For me the the variant of "if that is there and this is not do x; elif if this is there and that is not do y" is a bit more intuitive, but as both work, I'm fine with it.